paperslicer/grobid/figures.py

```
@@ -0,0 +1,121 @@
+        return ""
+    return " ".join("".join(node.itertext()).split()).strip()
+
+
+def _build_zone_index(root: etree._Element) -> Dict[str, Tuple[int, List[float]]]:
+    """Map zone xml:id -> (page, [x1,y1,x2,y2]) using facsimile/surface/zone ulx/uly/lrx/lry."""
+    idx: Dict[str, Tuple[int, List[float]]] = {}
+    for surf in root.findall(".//tei:facsimile/tei:surface", TEI_NS):
+        page_num = surf.get("n")
+        try:
+            page = int(page_num) if page_num else None
+        except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)